app/presenters/nsm/incomplete_items.rb

```diff
@@ -0,0 +1,78 @@
+module Nsm
+  class IncompleteItems
+    include ActionView::Helpers::TagHelper
+    include ActionView::Helpers::TranslationHelper
+
+    # govuk_link_to required modules
+    include GovukLinkHelper
+    include GovukVisuallyHiddenHelper
+    include ActionView::Helpers::UrlHelper
+    # for path methods
+    include Routing
+
+    attr_reader :controller
+
+    EXPECTED_ITEM_TYPES = [:work_items, :disbursements].freeze
+
+    def initialize(claim, type, controller)
+      raise "Invalid item type: #{type}" unless type.in?(EXPECTED_ITEM_TYPES)
+
+      @claim = claim
+      @type = type
+      @controller = controller
+    end
+
+    def incomplete_items
+      items.reject(&:complete?).sort_by(&:position)
+    end
+
+    def summary
+      "#{t("#{path_key}.incomplete_summary", count: incomplete_items.count)}: #{links}"
+    end
+
+    def links
+      incomplete_items.map do |item|
+        govuk_link_to(
+          path_title(item),
+          path_url(item)
+        )
+      end.join(', ')
+    end
+
+    private
+
+    def items
+      if @type == :work_items
```

Comment:
```ruby
@items ||= case @type
           when :work_items then @claim.work_items 
           when :disbursements then @claim.disbursements
           end
```

No reason you can't just set this in the constructor though

Reply:
I was using case like this for it in a previous iteration but it didn't remedy the coverage issues, but yeah thinking about it probably better format wise

I thought since if we have the type in the constructor, then we infer the items from the type, it would reduce the risk of us having code where someone mismatches the type to the items they feed in. Or do you mean creating the instance variable within the initializer instead of a private method?

Reply:
Yeah just creating it in the constructor. You can have the fallback case just throw the error and then you're covered